components/accelerator/nvidia/error/xid/reason.go

```diff
@@ -0,0 +1,43 @@
+package xid
+
+import (
+	"encoding/json"
+
+	nvidia_query_nvml "github.com/leptonai/gpud/components/accelerator/nvidia/query/nvml"
+)
+
+// Reason defines the reason for the output evaluation in the JSON format.
+type Reason struct {
+	// Messages are the messages for the reason.
+	// And do not include the errors.
+	Messages []string `json:"messages"`
+
+	// Errors are the xid errors that happened, keyed by the XID.
+	Errors map[uint64]XidError `json:"errors"`
+
+	// OtherErrors are other errors that happened during the evaluation.
+	OtherErrors []string `json:"other_errors,omitempty"`
+}
+
+func (r Reason) JSON() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+// XidError represents an XID error in the reason.
+type XidError struct {
+	// DataSource is the source of the data.
+	DataSource string `json:"data_source"`
+
+	// RawEvent is the raw event from the nvml API.
+	RawEvent nvidia_query_nvml.XidEvent `json:"raw_event"`
```

Comment:
Maybe dont need to pass whole rawEvent back to control plane, would make body too big, can just extract several necessary fields to XidError struct, like
SuggestedActions *common.SuggestedActions `json:"suggested_actions,omitempty"`
Description            string `json:"description"`
DeviceUUID string `json:"device_uuid"`